packages/core/admin/admin/src/pages/SettingsPage/pages/Roles/EditPage/tests/index.test.js

```diff
@@ -22,34 +23,47 @@ jest.mock('@strapi/helper-plugin', () => ({
   useOverlayBlocker: jest.fn(() => ({ lockApp: jest.fn(), unlockApp: jest.fn() })),
 }));
 
-const makeApp = (history) => (
-  <IntlProvider
-    messages={{ 'Settings.roles.form.created': 'created' }}
-    textComponent="span"
-    locale="en"
-    defaultLocale="en"
-  >
-    <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
-      <Theme>
-        <Router history={history}>
-          <Switch>
-            <Route path="/settings/roles/:id">
-              <EditPage />
-            </Route>
-          </Switch>
-        </Router>
-      </Theme>
-    </ThemeToggleProvider>
-  </IntlProvider>
-);
+const setup = (props) =>
+  render(<EditPage {...props} />, {
+    wrapper({ children }) {
+      const history = createMemoryHistory({
+        initialEntries: ['/settings/roles/1'],
+      });
+      const client = new QueryClient({
+        defaultOptions: {
+          queries: {
+            retry: false,
+          },
+        },
+      });
+
+      return (
+        <QueryClientProvider client={client}>
+          <IntlProvider
+            messages={{ 'Settings.roles.form.created': 'created' }}
+            textComponent="span"
+            locale="en"
+            defaultLocale="en"
+          >
+            <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
+              <Theme>
+                <Router history={history}>
```

Comment:
i'd use `MemoryRouter` here and instead pass `initialEntries` – I think thats the prop name for what you're doing here :) it's inline with the (slow) direction of refactoring some tests

Reply:
What do you think about deleting the whole test instead? Right now this setup for a snapshot test that proves the loading is rendered, which imo has 0 zero value.

Reply:
Y E S

Reply:
no false confidence